src/TypeScript/Rules/RuleDefinitionProviders.cs

```diff
@@ -28,6 +28,11 @@ internal interface ITypeScriptRuleDefinitionsProvider
         /// Returns the metadata descriptions for all TypeScript rules
         /// </summary>
         IEnumerable<RuleDefinition> GetDefinitions();
+
+        /// <summary>
+        /// Returns the rule key to display in VS for the specified ESLint rule
+        /// </summary>
+        string GetSonarRuleKey(string eslintRuleKey);
```

Comment:
Moved the single method from the key mapper interface to this class.
It seems reasonable that the class that is the source of the rule definitions should know how to map between the keys.